src/bindings/python/tests/test_runtime/test_tensor.py

```diff
@@ -148,6 +149,62 @@ def test_init_with_numpy_copy_memory(ov_type, numpy_dtype):
     assert ov_tensor.byte_size == arr.nbytes
 
 
+def test_init_with_node_output_port():
+    param1 = ops.parameter(ov.Shape([1, 3, 2, 2]), dtype=np.float64)
+    param2 = ops.parameter(ov.Shape([1, 3, 32, 32]), dtype=np.float64)
+    param3 = ops.parameter(ov.PartialShape.dynamic(), dtype=np.float64)
+    ones_arr = np.ones(shape=(1, 3, 32, 32), dtype=np.float64)
+    tensor1 = Tensor(param1.output(0))
+    tensor2 = Tensor(param2.output(0), ones_arr)
+    tensor3 = Tensor(param3.output(0))
+    with pytest.warns(RuntimeWarning):
+        tensor4 = Tensor(param3.output(0), ones_arr)
+    assert tensor1.shape == param1.shape
+    assert tensor1.element_type == param1.get_element_type()
+    assert tensor2.shape == param2.shape
+    assert tensor2.element_type == param2.get_element_type()
+    assert tensor3.shape == ov.Shape([0])
+    assert tensor3.element_type == param3.get_element_type()
+    assert tensor4.shape == ov.Shape([0])
+    assert tensor4.element_type == param3.get_element_type()
+
+
+def test_init_with_node_constoutput_port(device):
+    compiled_model = generate_relu_compiled_model(device)
+    output = compiled_model.output(0)
+    ones_arr = np.ones(shape=(1, 3, 32, 32), dtype=np.float32)
+
+    tensor1 = Tensor(output)
+    tensor2 = Tensor(output, ones_arr)
+
+    output_node = output.get_node()
+    assert tensor1.shape == output_node.shape
+    assert tensor1.element_type == output_node.get_element_type()
+    assert tensor2.shape == output_node.shape
+    assert tensor2.element_type == output_node.get_element_type()
+    assert np.array_equal(tensor2.data, ones_arr)
+
+
+def test_init_with_output_port_different_shapes():
+    param1 = ops.parameter(ov.Shape([2]), dtype=np.float32)
+    param2 = ops.parameter(ov.Shape([5]), dtype=np.float32)
+
+    ones_arr = np.ones(shape=(2, 2), dtype=np.float32)
+    with pytest.warns(RuntimeWarning):
+        Tensor(param1.output(0), ones_arr)
+
+    with pytest.raises(RuntimeError) as e:
+        Tensor(param2.output(0), ones_arr)
+    assert "Shape of the port exceeds shape of the array." in str(e.value)
+
+
+def test_init_with_output_port_different_types():
+    param1 = ops.parameter(ov.Shape([2, 2]), dtype=np.int8)
+    data = np.random.rand(2, 2).astype(np.float32)
+    with pytest.warns(RuntimeWarning):
+        Tensor(param1.output(0), data)
```

Comment:
A behavior of casting could be tested here.

Reply:
added